crates/trie/src/metrics.rs

```diff
@@ -0,0 +1,65 @@
+use crate::stats::TrieStats;
+use metrics::Histogram;
+use reth_metrics::Metrics;
+
+/// Wrapper for state root metrics.
+#[derive(Debug)]
+pub struct StateRootMetrics {
+    /// State trie metrics.
+    pub state_trie: TrieRootMetrics,
+    /// Storage trie metrics.
+    pub storage_trie: TrieRootMetrics,
+}
+
+impl Default for StateRootMetrics {
+    fn default() -> Self {
+        Self {
+            state_trie: TrieRootMetrics::new(TrieType::State),
+            storage_trie: TrieRootMetrics::new(TrieType::Storage),
+        }
+    }
+}
+
+/// Metrics for trie root calculation.
+#[derive(Clone, Metrics)]
+#[metrics(scope = "trie")]
+pub struct TrieRootMetrics {
+    /// The number of seconds trie root calculation lasted.
+    duration: Histogram,
+    /// The number of branches added during trie root calculation.
+    branches_added: Histogram,
+    /// The number of leaves added during trie root calculation.
+    leaves_added: Histogram,
```

Comment:
if the number of branch/leaf nodes can only grow, shouldn't we use counters here?

Reply:
these are the numbers of leaves and branches considered during the calculation. these numbers don't necessarily grow

Reply:
i meant the cumulative number of added nodes, but sgtm